### DISCUSSION TITLE
Help me correctly understand the credentials provider

### DISCUSSION THREAD

**Discussion by snax4a** [nextauthjs/next-auth#3963] [Help]
Hello,

I am a bit confused about the CredentialsProvider use case.

I have my custom API with a /get-token endpoint which i want use for credentials authentication.
If correct credentials are provided it returns this kind of response:
```json
{
    "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwOi8vc2NoZW1hcy54bWxz...",
    "refreshToken": "hdw+CVCmzLNF+tCgtykG9szl7wBE25G8SMTSFr5ZRZI=",
    "refreshTokenExpiryTime": "2022-02-19T14:10:23.911142Z"
}
```

I decided to use next-auth credentials provider and I am hitting this endpoint from the authorize method and returning these tokens as a user.

But in all the tutorials and examples I see that authorize method is used to return user object and If I understand correctly the JWT is generated by the next-auth.

Is It possible to use next-auth and disable JWT generation on next-auth but generate the JWT on the external API and only store it in the next-auth session?

In the JWT I have user details encrypted so I can return them also from authorize but I wonder if its necessary If I want next-auth to only hold my tokens but the user profile I can fetch with react-query.

How can I disable next-auth jwt encode and decode methods as I want to use the token returned from the API?

Unfortunately, I could not find any similar next-auth example with react-query

**Comment by ricardo-rp** [comment]
@snax4a I'm also trying to use next-auth with an existing api and hit this wall.
My guess is that they intend you to use this when _migrating_ from an existing API to a fullstack nextjs project?
Would like to know if anyone's found a solution.
